runtime/gc_ctrl.c

```diff
@@ -643,18 +640,20 @@ void caml_init_gc (uintnat minor_size, uintnat major_size,
                    uintnat custom_maj, uintnat custom_min,
                    uintnat custom_bsz)
 {
-  uintnat major_heap_size =
-    Bsize_wsize (caml_normalize_heap_increment (major_size));
```

Comment:
If I read you correctly, `caml_normalize_heap_increment` was used incorrectly, because it expected a size in bytes and returned a size in bytes, while here it is given a size in words and is expected to return a size in words.

Reply:
Indeed.